File: CardGame.Application/DTOs/PlayerHandInfoDto.cs
namespace CardGame.Application.DTOs;

/// <summary>
/// Represents information about a player within the PlayerGameStateDto.
/// Hand card details are omitted here; count is shown for all players.
/// </summary>
public class PlayerHandInfoDto 
{
    public Guid PlayerId { get; set; }
    public string Name { get; set; } = string.Empty;
    public int Status { get; set; }
    public int HandCardCount { get; set; } // Show count for all players
    public List<CardDto> PlayedCards { get; set; } = new List<CardDto>();
    public int TokensWon { get; set; }
    public bool IsProtected { get; set; }
}

File: CardGame.Web/ClientApp/src/app/core/models/playerHandInfoDto.ts
import { CardDto } from './cardDto';

/**
 * Represents information about a player within the PlayerGameStateDto.
 */
export interface PlayerHandInfoDto {
  playerId: string; // Changed from Guid
  name: string;
  status: number; // PlayerStatus enum value (e.g., 1=Active, 2=Eliminated)
  handCardCount: number; // Show count for all players
  playedCards?: CardDto[];
  tokensWon: number;
  isProtected: boolean;
}
